pkg/operator/controllers/guardrails/policies/gktemplates/aro-deny-clusterversion.yaml

```diff
@@ -0,0 +1,190 @@
+apiVersion: templates.gatekeeper.sh/v1
+kind: ConstraintTemplate
+metadata:
+  name: arodenyclusterversion
+  annotations:
+    metadata.gatekeeper.sh/title: "ClusterVersion"
+    metadata.gatekeeper.sh/version: 1.0.0
+    description: >-
+      Disallows editing ClusterVersion by regular users if there is a "cloud.openshift.com" entry in "openshift-config/pull-secret".
+
+spec:
+  crd:
+    spec:
+      names:
+        kind: ARODenyClusterVersion
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          type: object
+          description: >-
+            Disallows editing ClusterVersion by regular users if there is a "cloud.openshift.com" entry in "openshift-config/pull-secret".
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package arodenyclusterversion
+
+        import data.lib.common.is_exempted_account
+        import future.keywords.in
+
+        # Use object
+        # To retrieve from a different resource, data.inventory.namespace["openshift-managed-upgrade-operator"]["v1"]["ConfigMap"]["managed-upgrade-operator-config"]["data"]["config.yaml"]
+        violation[{"msg": msg}] {
+        	input.review.operation in ["CREATE", "UPDATE", "DELETE"]
+
+        	# ## Check user type
+        	not is_exempted_account(input.review)
+
+        	# ## If regular user and
+        	# ## has NO cloud.openshift.com entry in openshift-config/pull-secret Secret
+        	# ## ALLOW EDITING
+
+        	# ## If regular user and 
+        	# ## HAS cloud.openshift.com entry (`source: OCM` indicates pull-secret exists) in openshift-config/pull-secret Secret
+        	# ## NOT ALLOWED
+        	config_data := data.inventory.namespace["openshift-managed-upgrade-operator"]["v1"]["ConfigMap"]["managed-upgrade-operator-config"]["data"]["config.yaml"]
+        	regex.match("source: OCM", config_data)
```

Comment:
ok, looks like this template is quite identical to the `aro-deny-upgradeconfig` one, so my understanding is, we need two different constraints to facilitate different matches, but can use reuse the template at least?